docs/tasks/administer-cluster/kms-provider.md

```diff
@@ -0,0 +1,182 @@
+---
+approvers:
+- smarterclayton
+title: Using a KMS provider for data encryption
+---
+{% capture overview %}
+This page shows how to configure a Key Management Service (KMS) provider and plugin to enable secret data encryption.
+{% endcapture %}
+
+{% capture prerequisites %}
+
+* {% include task-tutorial-prereqs.md %}
+
+* Kubernetes version 1.10.0 or later is required
+
+* etcd v3 or later is required
+
+*The KMS provider is alpha in Kubernetes version 1.10.0, which means that it may change without notice. You may be required to decrypt your data prior to upgrading to 1.11.0
+
+{% endcapture %}
+
+{% capture steps %}
+
+The KMS encryption provider uses an envelope encryption scheme to encrypt data in etcd. The Key encryption keys (KEKs) are 
+stored and managed in a remote KMS. The KMS provider uses gRPC to communicate with a specific KMS 
+plugin. The KMS plugin, which is implemented as a gRPC server and deployed on the same host(s) as the Kubernetes master(s), is responsible for all communication with the remote KMS.
+
+## Configuring the KMS provider
+
+To configure a KMS provider on the API server, include a provider of type kms in the providers array in the encryption configuration file and set the following properties:
+
+  * `name`: Display name of the KMS plugin.
+  * `endpoint`: Listen address of the gRPC server (KMS plugin). The endpoint is a UNIX domain socket connection.
+  * `cachesize`: Number of data encryption keys (DEKs) to be cached in the clear. DEKs are used in Envelope encryption. 
+
+See [Understanding the encryption at rest configuration.](/docs/tasks/administer-cluster/encrypt-data)
+
+## Implementing a KMS plugin
+
+To implement a KMS plugin, you can develop a new plugin gRPC server or enable a KMS plugin already provided by your cloud provider. You then integrate the plugin with the remote KMS and deploy it on the Kubernetes master.
```

Comment:
Alright, "implement" here can mean "developing a new plugin" or "enabling an existing one" ...
I was confused by the verb used.